backend-shared/hack/service_account.go

```diff
@@ -0,0 +1,170 @@
+				if !apierr.IsAlreadyExists(err) {
+					_, err := clientset.RbacV1().Roles(namespace).Update(context.Background(), &role, metav1.UpdateOptions{})
+					if err != nil {
+						return "", err
+					}
+				}
+			}
+
+			roleBinding := rbacv1.RoleBinding{
+				TypeMeta: metav1.TypeMeta{
+					APIVersion: "rbac.authorization.k8s.io/v1",
+					Kind:       "RoleBinding",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name: ArgoCDManagerClusterRoleBinding,
+				},
+				RoleRef: rbacv1.RoleRef{
+					APIGroup: "rbac.authorization.k8s.io",
+					Kind:     "Role",
+					Name:     ArgoCDManagerClusterRole,
+				},
+				Subjects: []rbacv1.Subject{{
+					Kind:      rbacv1.ServiceAccountKind,
+					Name:      ArgoCDManagerServiceAccount,
+					Namespace: ns,
+				}},
+			}
+			_, err = clientset.RbacV1().RoleBindings(namespace).Create(context.Background(), &roleBinding, metav1.CreateOptions{})
+			if err != nil {
+				if !apierr.IsAlreadyExists(err) {
+					_, err := clientset.RbacV1().RoleBindings(namespace).Update(context.Background(), &roleBinding, metav1.UpdateOptions{})
+					if err != nil {
+						return "", err
+					}
+				}
+			}
+		}
+	}
+	return GetServiceAccountBearerToken(clientset, ns, ArgoCDManagerServiceAccount)
+}
+
+// GetServiceAccountBearerToken will attempt to get the provided service account until it
+// exists, iterate the secrets associated with it looking for one of type
+// kubernetes.io/service-account-token, and return it's token if found.
+func GetServiceAccountBearerToken(clientset kubernetes.Interface, ns string, sa string) (string, error) {
+	var serviceAccount *corev1.ServiceAccount
+	var secret *corev1.Secret
+	var err error
+	err = wait.Poll(500*time.Millisecond, 30*time.Second, func() (bool, error) {
+		serviceAccount, err = clientset.CoreV1().ServiceAccounts(ns).Get(context.Background(), sa, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		// Scan all secrets looking for one of the correct type:
+		for _, oRef := range serviceAccount.Secrets {
+			var getErr error
+			secret, err = clientset.CoreV1().Secrets(ns).Get(context.Background(), oRef.Name, metav1.GetOptions{})
+			if err != nil {
+				return false, fmt.Errorf("Failed to retrieve secret %q: %v", oRef.Name, getErr)
+			}
+			if secret.Type == corev1.SecretTypeServiceAccountToken {
+				return true, nil
+			}
+		}
+		return false, nil
+	})
+	if err != nil {
+		return "", fmt.Errorf("Failed to wait for service account secret: %v", err)
+	}
+	token, ok := secret.Data["token"]
+	if !ok {
+		return "", fmt.Errorf("Secret %q for service account %q did not have a token", secret.Name, serviceAccount)
+	}
+	return string(token), nil
+}
+
+func generateClientFromClusterServiceAccount(hostname string, bearerToken string) client.Client {
+	config := &rest.Config{
+		Host:        hostname,
+		BearerToken: bearerToken,
+	}
+	clientObj, err := client.New(config, client.Options{Scheme: scheme.Scheme})
+	if err != nil {
+		log.Fatal(err)
+		return nil
```

Comment:
Make sure that in all of your `return` statements there is a blank line _before_ them, to emphasise them.

Reply:
I'm not sure I agree with this as a stylistic format, and thus don't think it should be a requirement as a change to this PR.

Reply:
Yes, there's absolutely no need change the PR for such stuff, sorry if I didn't make it clear enough in my review. Honestly, I am also not sure I agree with some of these stylistic choices. I am just used to use https://github.com/bombsimon/wsl in my IDE, which is something I would be happy to get rid of :D